extension/src/experiments/model/util.ts

```diff
@@ -1,40 +1,38 @@
 import get from 'lodash.get'
 import { formatDate } from '../../util/date'
-import { truncate } from '../../util/string'
 import { splitColumnPath } from '../columns/paths'
 import { Experiment } from '../webview/contract'
 
 type Value = undefined | null | [] | string | number
 
-const getValueType = (value: Value): string => {
-  let type: string = typeof value
+const isDate = (value: Value): boolean =>
+  !!(typeof value === 'string' && Date.parse(value))
+
+const isLongNumber = (value: Value): boolean =>
+  typeof value === 'number' && value.toString().length > 7
+
+const getStringifiedValue = (value: Value): string => {
   if (Number.isNaN(value)) {
-    type = 'NaN'
-  } else if (typeof value === 'string' && Date.parse(value)) {
-    type = 'date'
-  } else if (Array.isArray(value)) {
-    type = 'array'
-  } else if (value === '') {
-    type = 'empty'
+    return 'NaN'
   }
-  return type
-}
 
-const getStringifiedValue = (value: Value): string => {
-  const type = getValueType(value)
-  switch (type) {
-    case 'date':
-      return typeof value === 'string' ? formatDate(value) : ''
-    case 'array':
-      return `[${value?.toString()}]`
-    case 'empty':
-    case 'NaN':
-      return type
-    case 'number':
-      return truncate(String(value), 7)
-    default:
-      return String(value)
+  if (isDate(value)) {
+    return formatDate(value as string)
+  }
+
+  if (Array.isArray(value)) {
+    return `[${value?.toString()}]`
+  }
+
+  if (value === undefined) {
+    return '-'
   }
+
+  if (isLongNumber(value)) {
+    return (value as number).toPrecision(5)
```

Comment:
I think it might have some implications:

just looking briefly into this - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Number/toPrecision

it's doing a lot of stuff under the hood?

also, I would check how does it behave with just a regular large number - `12345678900`? Will it transform it into scientific notation? (may be it's even better, btw vs just throwing part of the number away).

Suggestion- let's do this as a separate PR? do a proper research what do we want to show for different types of numbers? wdyt?

Reply:
For now I copied the code that we use to display numbers in the table: 

```
const defaultPrecision = 5 // for when we can't calculate real precision yet

export const formatFloat = (value: number): string => {
  const automatic = value.toString()
  if (automatic.length > 7) {
    return value.toPrecision(defaultPrecision)
  }
  return automatic
}
```

Whilst not perfect I thought this was better than truncating the number.

Reply:
It will transform large numbers into scientific notation. I also spent time researching alternatively yesterday and could not come up with anything better, which is why I proposed this.

Reply:
sounds good, Matt, thanks